### PR TITLE
Add CommaTheBot

### LINKED ISSUE #202: Bot changing "foo, the" to "the foo"
See README in CommaTheBot/

### PR DESCRIPTION
**Pull request by bsvka**
Closes #202

## Description:
In this Pull Request we have made the following changes:
* Added CommaTheBot

### CODE REVIEW

**Comment by bsvka** [comment]
I would also like to apply for `bot` privileges on the account [CommaTheBot](https://openlibrary.org/people/commathebot).
@mekarpeles @hornc 
Thanks.

File: CommaTheBot/CommaTheBot.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            '[Tt]he',
+            '[Dd]er', '[Dd]ie', '[Dd]as',
+            '[Ll]e', '[Ll]a', '[Ee]l',
+            '[Ll]os', '[Ll]as', '[Ll]es',
+        ]
+
+        self.pattern = re.compile(rf"^([\w ,]*), ?({'|'.join(articles)})$")
+
+    def needs_fixing(self, edition_title: str) -> bool:
+        if edition_title == None: return False # no title given
```
Comment (cclauss) [inline CommaTheBot/CommaTheBot.py:24]:
```suggestion
        return edition_title is not None
```

Comment (cclauss) [inline CommaTheBot/CommaTheBot.py:24]: See PEP8 discussion about comparing to ˋNone`.

Comment (bsvka) [reply CommaTheBot/CommaTheBot.py]: Addressed in 0a67693, thank you.

**Review by cclauss — CHANGES_REQUESTED** [review]

File: CommaTheBot/main.sh [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+pip install -r requirements.txt
+
+# dl latest dump
+curl https://openlibrary.org/data/ol_dump_latest.txt.gz -O
+
+# prefilter it
+file=( ol_dump*.gz )
+zgrep -E "^/type/(?:edition|work)" "$file" | pv | gzip > "filtered_$file"
+
+python CommaTheBot.py --file "filtered_$file" "$@"
```
Comment (cclauss) [inline CommaTheBot/main.sh:20]:
```suggestion
python CommaTheBot.py --file "filtered_$file" "$@"

```

File: CommaTheBot/test_CommaTheBot.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        title = "Foo, The "
+        assert commaTheBot.needs_fixing(title) == False
+
+        # maybe support for these should be added
+        title = "Foo (bar), The"
+        assert commaTheBot.needs_fixing(title) == False
+        title = "Foo., The"
+        assert commaTheBot.needs_fixing(title) == False
+        title = "foo-bar, The"
+        assert commaTheBot.needs_fixing(title) == False
```
Comment (cclauss) [inline CommaTheBot/test_CommaTheBot.py:62]:
```suggestion
        assert commaTheBot.needs_fixing(title) == False

```

**Review by cclauss — CHANGES_REQUESTED** [review]

**Pull request merged**